lib/cc/engine/flay.rb

```diff
@@ -0,0 +1,57 @@
+require 'flay'
+require 'json'
+
+module CC
+  module Engine
+    class Flay
+      def initialize(directory: ,engine_config: ,io:)
+        @directory = directory
+        @engine_config = engine_config
+        @io = io
+      end
+
+      def run
+        files = ::Flay.filter_files(::Flay.expand_dirs_to_files(directory))
+        flay.process(*files)
+
+        flay.report(StringIO.new).each do |issue|
+          puts "#{new_violation(issue).to_json}\n"
```

Comment:
These should be separated with `\0`, instead of `\n`. JSON can contain newlines so we use the null byte to be safer.